src/lib/components/common/RemoveItemButton.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { 
+    Grid,
+    IconButton,
+    Tooltip,
+} from '@material-ui/core';
+import RemoveCircle from '@material-ui/icons/RemoveCircle';
+
+/*
+    Displays a "-" icon and can be used to remove instances of list-type items.
+    It is passed a handler function as a prop, which should remove the specified
+    instance from the list of items being updated. It is also passed an index
+    value as a prop, which should be used to identify the specific object
+    instance to be removed.
+*/
+const RemoveItemButton = (props) => {
+    return (
+        <div>
+            {props.display
+                ?
+                    <Grid item xs={1}>
```

Comment:
In some cases this component is not vertically centred relative to the text fields (eg. Checksums, Local File Access Objects, and AWS S3 Access Objects).